modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java

```diff
@@ -181,17 +185,22 @@ protected void createResourceBundle(Locale locale) {
 			portletResourceBundle, portalResourceBundle);
 	}
 
-	protected void evaluateDDMFormRule(DDMFormRule ddmFormRule) {
+	protected void evaluateDDMFormRule(
+		DDMFormRule ddmFormRule, List<String> actionsAlreadyEvaluated) {
+
 		if (evaluateDDMFormRuleCondition(ddmFormRule.getCondition())) {
 			List<String> actions = ddmFormRule.getActions();
 
 			Stream<String> stream = actions.stream();
 
 			evaluateDDMFormRuleAction(
 				stream.collect(Collectors.joining(" AND ")));
+
+			actionsAlreadyEvaluated.add(actions.toString());
 		}
 		else {
-			_ddmFormRuleHelper.checkFieldAffectedByAction(ddmFormRule);
+			_ddmFormRuleHelper.checkFieldAffectedByAction(
+				ddmFormRule, actionsAlreadyEvaluated);
```

Comment:
Hi @gabrielibson, from what I could see, the main part of your fix is on the lines above, therefore I believe that we can optimize your fix creating a private `List` on `DDMFormEvaluatorHelper` to store the already evaluated actions and then check it before call the method `_ddmFormRuleHelper.checkFieldAffectedByAction`. A possible way to do this can be cloning the `ddmFormRule`, iterating through `ddmFormRule.getActions()` and filtering these rules according to the fix needs.

What do you think about this suggestion? Do you think that it's feasible?

If you don't have time to work on this issue anymore due to your new duties, please let me know. 😉

Reply:
Hi @gabrielibson, did you have the chance to check my previous comment above?

Reply:
Hi @natocesarrego, yes I did. I had to deal with some environment issues to come back to this PR, but now is everything ok, so I'll work on it as soon as possible. I believe that until tomorrow it will be done.

Thanks.

Reply:
Ok, @gabrielibson. Thanks for your update. 😉